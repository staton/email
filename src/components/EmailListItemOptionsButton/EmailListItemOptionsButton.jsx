import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    className: PropTypes.string,
    content: PropTypes.any,
    onClick: PropTypes.func
};

const defaultProps = {
    className: null,
    content: null,
    onClick: () => {}
};

class EmailListItemOptionsButton extends Component {

	render() {
		return (
            <button
                className={this.getClassName()}
                onClick={this.props.onClick}
            >
                {this.props.content}
            </button>
		);
    }

    /**
     * Gets the class name(s) for this component.
     * @returns {string} The class name(s).
     */
    getClassName() {
        let className = 'EmailListItemOptionsButton';

        if (this.props.className) {
            className += ' ' + this.props.className;
        }

        return className;
    }

}

EmailListItemOptionsButton.propTypes = propTypes;
EmailListItemOptionsButton.defaultProps = defaultProps;

export default EmailListItemOptionsButton;