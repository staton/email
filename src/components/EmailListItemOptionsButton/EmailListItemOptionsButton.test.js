import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import EmailListItemOptionsButton from './EmailListItemOptionsButton';

describe('<EmailListItemOptionsButton />', () => {

    it('renders correctly', () => {
        const wrapper = shallow(
            <EmailListItemOptionsButton
                className="EmailListItemOptionsButton__delete"
                content="X" 
            />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('handles click events', () => {
        const onClick = jest.fn();
        const wrapper = mount(
            <EmailListItemOptionsButton
                className="EmailListItemOptionsButton__delete"
                content="X" 
                onClick={onClick}
            />);
        const icon = wrapper.find('.EmailListItemOptionsButton');
        icon.simulate('click');
        expect(onClick).toHaveBeenCalledTimes(1);
    });
    
});