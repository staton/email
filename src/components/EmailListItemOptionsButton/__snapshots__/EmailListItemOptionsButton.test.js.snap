// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailListItemOptionsButton /> renders correctly 1`] = `
<button
  className="EmailListItemOptionsButton EmailListItemOptionsButton__delete"
  onClick={[Function]}
>
  X
</button>
`;
